src/Glpi/Http/LegacyRouterListener.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Glpi\Http;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final readonly class LegacyRouterListener implements EventSubscriberInterface
+{
+    use LegacyRouterTrait;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => ['onKernelRequest', 260],
+        ];
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $request = $event->getRequest();
+
+        $response = $this->runLegacyRouter($request);
+
+        if ($response) {
+            $event->setResponse($response);
+        }
+    }
+
+    private function runLegacyRouter(Request $request): ?Response
+    {
+        /**
+         * GLPI web router.
+         *
+         * This router is used to be able to expose only the `/public` directory on the webserver.
+         */
+
+        $glpi_root = \dirname(__DIR__, 3);
+
+        if (
+            $request->server->get('SCRIPT_NAME') === '/public/index.php'
+            && preg_match('/^\/public/', $request->server->get('REQUEST_URI')) !== 1
+        ) {
+            // When requested URI does not start with '/public' but `$request->server->get('SCRIPT_NAME')` is '/public/index.php',
+            // it means that document root is the GLPI root directory, but a rewrite rule redirects the request to the PHP router.
+            // This case happen when redirection to PHP router is made by an `.htaccess` file placed in the GLPI root directory,
+            // and has to be handled to support shared hosting where it is not possible to change the web server root directory.
+            $uri_prefix = '';
+        } else {
+            // `$request->server->get('SCRIPT_NAME')` corresponds to the script path relative to server document root.
+            // -> if server document root is `/public`, then `$request->server->get('SCRIPT_NAME')` will be equal to `/index.php`
+            // -> if script is located into a `/glpi-alias` alias directory, then `$request->server->get('SCRIPT_NAME')` will be equal to `/glpi-alias/index.php`
+            $uri_prefix = rtrim(str_replace('\\', '/', dirname($request->server->get('SCRIPT_NAME'))), '/');
+        }
+
+        // Get URI path relative to GLPI (i.e. without alias directory prefix).
+        $path = preg_replace(
+            '/^' . preg_quote($uri_prefix, '/') . '/',
+            '',
+            parse_url($request->server->get('REQUEST_URI') ?? '/', PHP_URL_PATH)
+        );
```

Comment:
It may probably be moved in the trait, to prevent duplication with the `LegacyAssetsListener`.